Source/ListItemFormatter.swift

```diff
@@ -24,60 +24,116 @@ import Foundation
-        case standard, or, unit
+
+        /// Mode for a typical 'and' list of arbitrary placeholders.
+        case standard
+
+        /// Mode for a typical 'or' list of arbitrary placeholders.
+        case or
+
+        /// A format mode suitable for arbitrary placeholder units with no mode (i.e
+        ///  measurement units).
+        case unit
     }
 
-    /// TBC
+    /// The following constants specify predefined format styles for list items.
     @objc(LNListItemFormatterStyle) public enum Style: Int {
-        case `default`, narrow, short
+
+        /// A style suitable for regular values and wide units.
+        case `default`
+
+        /// A shorter style typically suitable for short or abbreviated placeholder
+        ///  values.
+        case short
+
+        /// The narrowest style where space on the screen is very limited. This style
+        ///  will only have an effect when used with the `.unit` mode.
+        case narrow
     }
 
-    /// TBC
+    /// The format mode of the receiver.
+    ///
+    /// The default value is set to `.standard`.
     @objc public var mode: Mode
 
-    /// TBC
+    /// The format style of the reciever.
+    ///
+    /// The default value is set to `.default`.
     @objc public var style: Style
 
-    /// TBC
+    /// The locale for the receiver.
+    ///
+    /// The default value is set to `Locale.autoupdatingCurrent`.
     @objc public var locale: Locale
 
-    /// TBC
+    /// The default text attributes used to display the output string.
+    ///
+    /// The value of this property is used by the receiver to produce the output
+    ///  `NSAttributedString` object of `ListItemFormatter.attributedString(from:)`.
+    ///
+    /// The default value is an empty dictionary.
     @objc public var defaultAttributes: [NSAttributedString.Key: Any] = [:]
 
-    /// TBC
+    /// The item specific attributes used to display the list items within the output
+    ///  string.
+    ///
+    /// The value of this property is used by the receiver to produce the output
+    ///  `NSAttributedString` object of `ListItemFormatter.attributedString(from:)`,
+    ///  specifically by applying these attributes only to the items within the `list`
+    ///  array. These attributes are applied on top of `defaultAttributes`.
+    ///
+    /// The default value of this property is an empty dictionary.
     @objc public var itemAttributes: [NSAttributedString.Key: Any] = [:]
 
-    /// TBC
+    /// Returns a string representation of the given list items formatted using the
+    /// receiver’s current settings.
+    ///
+    /// - Parameter list: The list items to be formatted into a localised, human
+    ///  readable list.
     ///
-    /// - Parameter list: TBC
-    /// - Returns: TBC
+    /// - Returns: A string representation of `list` formatted using the receiver’s
+    ///  current settings.
     @objc public func string(from list: [String]) -> String? {
         return string(for: list)
     }
```

Comment:
[Q] Do you think it's a good idea to overload the method in Swift and expose a different name to Objective-C?

```swift
@objc(stringFrom:)
func string(for list: [String]) -> String?
```

Reply:
Ahh I didn't think about this. I already opened #17 for improvements to the return value so can add this to that issue and tackle it separately to the documentation, thanks!